src/service/storage-provider/storage-provider.service.ts

```diff
@@ -31,76 +33,62 @@ export class StorageProviderService {
   ) {}
 
-  public async getProviderClients(
+  public async getProviderClientsWeekly(
     isAccumulative: boolean,
   ): Promise<HistogramWeekResponse> {
-    const clientProviderDistributionWeeklyTable = isAccumulative
-      ? Prisma.ModelName.client_provider_distribution_weekly_acc
-      : Prisma.ModelName.client_provider_distribution_weekly;
-
-    const providerCountResult = await this.prismaService.$queryRaw<
-      [
-        {
-          count: number;
-        },
-      ]
-    >`select count(distinct provider)::int
-      from ${modelName(clientProviderDistributionWeeklyTable)}`;
+    const providerCount = await this.getProviderCount();
 
     const query = isAccumulative
       ? getProviderClientsWeeklyAcc
       : getProviderClientsWeekly;
 
-    return await this.histogramHelper.getWeeklyHistogramResult(
-      await this.prismaService.$queryRawTyped(query()),
-      providerCountResult[0].count,
+    return new HistogramWeekResponse(
+      providerCount,
+      await this.histogramHelper.getWeeklyHistogramResult(
+        await this.prismaService.$queryRawTyped(query()),
+      ),
     );
   }
 
-  public async getProviderBiggestClientDistribution(
+  public async getProviderBiggestClientDistributionWeekly(
     isAccumulative: boolean,
   ): Promise<HistogramWeekResponse> {
-    const clientProviderDistributionWeeklyTable = isAccumulative
-      ? Prisma.ModelName.client_provider_distribution_weekly_acc
-      : Prisma.ModelName.client_provider_distribution_weekly;
-
-    const providerCountResult = await this.prismaService.$queryRaw<
-      [
-        {
-          count: number;
-        },
-      ]
-    >`select count(distinct provider)::int
-      from ${modelName(clientProviderDistributionWeeklyTable)}`;
+    const providerCount = await this.getProviderCount();
 
     const query = isAccumulative
       ? getProviderBiggestClientDistributionAcc
       : getProviderBiggestClientDistribution;
 
-    return await this.histogramHelper.getWeeklyHistogramResult(
-      await this.prismaService.$queryRawTyped(query()),
-      providerCountResult[0].count,
+    return new HistogramWeekResponse(
+      providerCount,
+      await this.histogramHelper.getWeeklyHistogramResult(
+        await this.prismaService.$queryRawTyped(query()),
+      ),
     );
   }
 
-  public async getProviderRetrievability(
+  public async getProviderCount(): Promise<number> {
+    return (
+      await this.prismaService.$queryRaw<
+        [{ count: number }]
+      >`select count(distinct provider)::int
+        from ${modelName(Prisma.ModelName.providers_weekly_acc)}`
+    )[0].count;
+  }
+
+  public async getProviderRetrievabilityWeekly(
     isAccumulative: boolean,
   ): Promise<RetrievabilityWeekResponse> {
-    const providersWeeklyTable = isAccumulative
-      ? Prisma.ModelName.providers_weekly_acc
-      : Prisma.ModelName.providers_weekly;
-
-    const providerCountAndAverageSuccessRate = await this.prismaService
```

Comment:
old bug? should count all providers, not only in this week (see where clause). but this field is not even being used anywhere in frontend